docs/dev/setup.rst

```diff
@@ -593,38 +639,108 @@ HTTP. We need to extend the configuration in the file
+
+.. warning::
+  If you want to run multiple docker-compose setups in parallel on one host you might have to modify
+  volume, container and network names!
+
+Getting Started with Docker-Compose Setups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To get started with one of the mentioned Docker-Compose Setups do the following:
+
+.. TODO: modify config part accordingly to config decision
 
 1. Install `docker <https://docs.docker.com/install/>`__ and `docker-compose <https://docs.docker.com/compose/install/>`__
-2. Configure the credentials in ``application-artemis.yml`` in the folder ``src/main/resources/config`` as described above
-3. Run ``docker-compose up``
-4. Go to http://localhost:9000
+2. ( Depending on the chosen setup it's necessary to configure the Artemis configs like ``application-local.yml``
+   in the folder ``src/main/resources/config`` as described in the section `Dockerfile <#dockerfile>`__.
+   The default setup ``docker-compose.yml`` should run without the default configurations, so no changes are required.)
+3. Run ``docker-compose up`` or ``docker-compose -f src/main/docker/<setup to be launched>.yml up``
+4. For Artemis instances go to http://localhost:8080 (http://localhost:9000 for the seperated server and client setup)
+
+Debugging with Docker
+^^^^^^^^^^^^^^^^^^^^^
+
+See the `Debugging with Docker <#docker-debugging>`__ section for detailed information.
+In all development docker-compose setups like ``artemis-dev-mysql.yml`` Java Remote Debugging is enabled by default.
 
-The client and the server will run in different containers. As Npm is
-used with its live reload mode to build and run the client, any change
-in the client’s codebase will trigger a rebuild automatically. In case
-of changes in the codebase of the server one has to restart the
-``artemis-server`` container via
-``docker-compose restart artemis-server``.
+Service, Container and Volume names
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-(Native) Running and Debugging from IDEs is currently not supported.
+Service names for the usage within docker-compose are kept short, like ``mysql``, to make it easier
+to use them in a CLI.
+
+Container and volume names are prepended with ``artemis-`` in order to not interfere with other container or volume
+names on your system.
 
 Get a shell into the containers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+.. tip::
+  To keep the documentation short, we will use the standard form of ``docker-compose COMMAND`` from this point on.
+  You can use the following commands also with the ``-f src/main/docker/<setup to be launched>.yml`` argument pointing
+  to a specific setup.
+
 -  app container:
-   ``docker exec -it $(docker-compose ps -q artemis-app) sh``
+   ``docker-compose exec artemis-app bash``
 -  mysql container:
-   ``docker exec -it $(docker-compose ps -q artemis-mysql) mysql``
+   ``docker-compose exec mysql bash`` or directly into mysql ``docker-compose exec mysql mysql``
```

Comment:
```suggestion
   ``docker compose exec mysql bash`` or directly into mysql ``docker-compose exec mysql mysql``
```

Reply:
:+1: OMG I haven't known about the GO implementation of docker compose :sweat_smile: Guess I'll adapt this everywhere in our project as it sounds to be faster than the Python implementation and also backwards compatible.